gaffer-core/operation/src/main/java/gaffer/export/Exporter.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gaffer.export;
+
+import gaffer.commonutil.iterable.CloseableIterable;
+import gaffer.operation.impl.export.ExportOperation;
+import gaffer.user.User;
+import gaffer.util.ExportUtil;
+
+/**
+ * An <code>Exporter</code> can store data of any kind and retrieve it with
+ * pagination.
+ */
+public abstract class Exporter {
+    public static final String SEPARATOR = "_";
+    private User user;
+    private String plainTextUserId;
+    private Long timestamp = System.currentTimeMillis();
+    private String key = ExportOperation.DEFAULT_KEY;
+
+    /**
+     * Initialises the export. This base method just stores the current user.
+     * Override this method (and call super) to add further initialisation.
+     *
+     * @param key    export key
+     * @param config configuration for the export (This will be an instance of a gaffer Store)
+     * @param user   the user who initiated the export
+     */
+    public void initialise(final String key, final Object config, final User user) {
```

Comment:
Could we template the class of the config object? Looking at the Gaffer exporter we cast it to a Store, if we template it, the child classes wouldn't require casting